pkg/middleware/request_tracing.go

```diff
@@ -0,0 +1,42 @@
+package middleware
+
+import (
+	"net/http"
+	"time"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	tlog "github.com/opentracing/opentracing-go/log"
+
+	"gopkg.in/macaron.v1"
+)
+
+func RequestTracing() macaron.Handler {
+	return func(res http.ResponseWriter, req *http.Request, c *macaron.Context) {
+		rw := res.(macaron.ResponseWriter)
+
+		var span opentracing.Span
+		opName := req.URL.Path
+		carrier := opentracing.HTTPHeadersCarrier(req.Header)
+
+		wireContext, err := opentracing.GlobalTracer().Extract(
+			opentracing.HTTPHeaders, carrier)
+		if err != nil {
+			span = opentracing.StartSpan(opName)
+		} else {
+			span = opentracing.StartSpan(opName, opentracing.ChildOf(wireContext))
+		}
```

Comment:
I think this is the standard way to do this:
```
span := tracer.StartSpan("HTTP "+macCtx.Req.Method+" "+path, ext.RPCServerOption(spanCtx))
```

note that RPCServerOption will create a new root span if needed, or correctly inherits from an existing span, if found.